kbfssync/semaphore_test.go

```diff
@@ -131,19 +131,19 @@ func TestCancel(t *testing.T) {
 	s := NewSemaphore()
 	require.Equal(t, int64(0), s.countForTest())
 
+	// Do this before spawning the goroutine to avoid a race on
+	// the count field of the result of callAcquire.
```

Comment:
No, I think I mean `callAcquire`, but rewrote the comment to hopefully be clearer. You're right about the cause of the race.

I think I'd rather have the invariant that `count` is less than `n` if an error is returned, but the 'staleness' of the value isn't obvious. Updated the function comment to call it out.